### PR TITLE
RELATED: RAIL-4158 properly integrate KPI edit mode

### PR DESCRIPTION
**Pull request by no23reason**
* simplify KPI execution definitions
* properly use the RenderModeAware with KPI
* add option to delete the KPI widget

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/widget/kpi/EditableDashboardKpi/EditableDashboardKpi.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                kpiDataStatus === "success"
+                    ? {
+                          seriesBy: compact([kpiDataResult!.primaryMeasure, kpiDataResult!.secondaryMeasure]),
+                          filters: kpiDataResult!.effectiveFilters,
+                      }
+                    : undefined,
+        },
+        [
+            kpiDataStatus,
+            kpiDataResult?.primaryMeasure,
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/widget/kpi/EditableDashboardKpi/EditableDashboardKpi.tsx:90]: Maybe  you could spread primaryMeasure , secondaryMeasure and effectiveFilters before its added to dependency array.

File: libs/sdk-ui-dashboard/src/presentation/widget/kpi/EditableDashboardKpi/EditableDashboardKpi.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            {() => {
+                if (kpiDataStatus === "loading" || kpiDataStatus === "pending") {
+                    return <LoadingComponent />;
+                }
+
+                if (kpiDataStatus === "error") {
+                    return <ErrorComponent message={kpiDataError!.message} />;
+                }
+
+                const kpiResult = !result?.dataView.totalCount[0]
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/widget/kpi/EditableDashboardKpi/EditableDashboardKpi.tsx:155]: Put this logic into utils

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8847/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10688/ : SUCCESS in 18m 00s

**Review by ivanmjartan — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2756

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2527/ : SUCCESS in 5m 26s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2826/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1655/ : SUCCESS in 17m 59s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10690/ : SUCCESS in 16m 13s

**Pull request merged**